src/testRunner.ts

```diff
@@ -5,13 +5,154 @@ import { disabledPrettyPrinter } from './printer';
 import * as outputWindow from './results-output/results-doc';
 import { NReplSession } from './nrepl';
 import * as cider from './nrepl/cider'
+import * as lsp from './lsp/types'
 import * as namespace from './namespace';
 import { getSession, updateReplSessionType } from './nrepl/repl-session';
 import * as getText from './util/get-text';
 
 let diagnosticCollection = vscode.languages.createDiagnosticCollection('calva');
 
-function reportTests(results: cider.TestResults[]) {
+// Given a Clojure namespace name, in the form `foo.bar-baz` find a URI for the
+// matching source file in the workflows. We do this by:
+// - replacing . with /
+// - replace - with _
+// - search for .clj or cljs files that match.
+async function uriForNamespace(namespace: string): Promise<vscode.Uri> {
+    const path = namespace.replace(/\./g, "/").replace(/-/g, "_");
+    const uris = await vscode.workspace.findFiles('**/' + path + ".{clj,cljs}", null, 1);
+    return uris[0];
+}
```

Comment:
This might be a bit brittle, especially in a monorepo. If we assume we have a REPL connection, it's probably better to rely on nrepl for this. Maybe factor the `getUriForCurrentNamespace` out of `results-doc.ts` moving it to `namespace.ts`:

https://github.com/BetterThanTomorrow/calva/blob/60bb98616b577a960f15473a821978b62782ca9b/src/results-output/results-doc.ts#L374-L375

Reply:
I've done this 👍